blocks/progress-bar-table/progress-bar-table.js

```diff
@@ -0,0 +1,85 @@
+function calculateRatioPost(elements) {
+  const sizel = elements.length;
+  let ratio1;
+  let ratio2;
+  if (sizel === 2) return [100, 0];
+  const childNodes1 = elements[1].childNodes;
+  const childNodes2 = elements[2].childNodes;
+  ratio1 = parseFloat(childNodes1[7].textContent.substring(0, childNodes1[7].textContent.indexOf(' ')));
+  ratio2 = parseFloat(childNodes2[7].textContent.substring(0, childNodes1[7].textContent.indexOf(' ')));
+  if (ratio2 > ratio1) {
+    ratio1 = Math.floor((ratio1 * 100) / ratio2);
+    ratio2 = 100;
+  } else {
+    ratio2 = Math.floor((ratio2 * 100) / ratio1);
+    ratio1 = 100;
+  }
+  return [ratio1, ratio2];
+}
+
+const callback = (entries) => {
+  entries.forEach((entry) => {
+    const elements = entry.target.querySelectorAll('.progress-bar-table > div');
+    const compareTheValues = calculateRatioPost(elements);
+    const toChangea = entry.target.querySelector('.progress-value-1');
+    toChangea.style.width = ('0%');
+    toChangea.style.width = (`${compareTheValues[0]}%`);
+    const toChangeb = entry.target.querySelector('.progress-value-2');
+    toChangeb.style.width = ('0%');
+    toChangeb.style.width = (`${compareTheValues[1]}%`);
+    if (entries[0].intersectionRatio === 0) {
+      toChangea.style.width = ('0%');
+      toChangeb.style.width = ('0%');
+    }
+  });
+};
+
+function addDynamic2(block) {
```

Comment:
- use speaking method name to indicate what's actually happening

for example:
`function addObserver(block) {`